src/structures/GuildScheduledEvent.js

```diff
@@ -0,0 +1,311 @@
+ * Represents a scheduled event in a {@link Guild}.
+ * @extends {Base}
+ */
+class GuildScheduledEvent extends Base {
+  constructor(client, data) {
+    super(client);
+
+    /**
+     * The id of the guild scheduled event
+     * @type {Snowflake}
+     */
+    this.id = data.id;
+
+    /**
+     * The id of the guild this guild scheduled event belongs to
+     * @type {Snowflake}
+     */
+    this.guildId = data.guild_id;
+
+    /**
+     * Whether the guild scheduled event has been deleted
+     * @type {boolean}
+     */
+    this.deleted = false;
+
+    this._patch(data);
+  }
+
+  _patch(data) {
+    /**
+     * The channel id in which the scheduled event will be hosted, or `null` if entity type is `EXTERNAL`
+     * @type {?Snowflake}
+     */
+    this.channelId = data.channel_id;
+
+    /**
+     * The id of the user that created this guild scheduled event
+     * @type {?Snowflake}
+     */
+    this.creatorId = data.creator_id;
+
+    /**
+     * The name of the guild scheduled event
+     * @type {string}
+     */
+    this.name = data.name;
+
+    if ('description' in data) {
+      /**
+       * The description of the guild scheduled event
+       * @type {?string}
+       */
+      this.description = data.description;
+    } else {
+      this.description ??= null;
+    }
+
+    /**
+     * The timestamp the guild scheduled event will start at
+     * @type {number}
+     */
+    this.scheduledStartTime = data.scheduled_start_time;
+
+    /**
+     * The timestamp the guild scheduled event will end at,
+     * or `null` if the event does not have a scheduled time to end
+     * @type {?number}
+     */
+    this.scheduledEndTime = data.scheduled_end_time;
+
+    /**
+     * The privacy level of the guild scheduled event
+     * @type {PrivacyLevel}
+     */
+    this.privacyLevel = GuildScheduledEventPrivacyLevels[data.privacy_level];
+
+    /**
+     * The status of the guild scheduled event
+     * @type {GuildScheduledEventStatus}
+     */
+    this.status = GuildScheduledEventStatuses[data.status];
+
+    /**
+     * The type of hosting entity associated with the scheduled event
+     * @type {GuildScheduledEventEntityType}
+     */
+    this.entityType = GuildScheduledEventEntityTypes[data.entity_type];
+
+    /**
+     * The id of the hosting entity associated with the scheduled event
+     * @type {?Snowflake}
+     */
+    this.entityId = data.entity_id;
+
+    if ('user_count' in data) {
+      /**
+       * The number of users who are subscribed to this guild scheduled event
+       * @type {?number}
+       */
+      this.userCount = data.user_count;
+    } else {
+      this.userCount ??= null;
+    }
+
+    if ('creator' in data) {
+      /**
+       * The user that created this guild scheduled event
+       * @type {?User}
+       */
+      this.creator = this.client.users._add(data.creator);
+    } else {
+      this.creator ??= this.client.users.resolve(this.creatorId);
+    }
+
+    if (data.entity_metadata) {
```

Comment:
Ditto

Reply:
What's the status on this? I started with a separate structure, so should I revert back to that or keep it just like threads?

Reply:
@vladfrangu this is the only thing that's left (i guess)

Reply:
I still think that it should be `gse.metadata.location`.